dags/iop/parser.py

```diff
@@ -100,3 +105,80 @@ def _get_extracted_arxiv_eprint_value(self, article):
             self.logger.error("The arXiv value is not valid.", dois=self.dois)
         except AttributeError:
             self.logger.error("No arXiv eprints found", dois=self.dois)
+
+    def _get_authors(self, article):
+        contrib_types = article.findall(
+            "front/article-meta/contrib-group/contrib[@contrib-type='author']"
+        )
+        authors = []
+        surname = ""
+        given_names = ""
```

Comment:
why you set them here?

Reply:
1. I would get an error of  _UnboundLocalError: local variable referenced before assignment:_ here:
https://github.com/cern-sis/workflows/blob/be9436b94e94a7ec581404bdb3562e64a4d70b03/dags/iop/parser.py#L137-L138
2. I need to have string here: https://github.com/cern-sis/workflows/blob/be9436b94e94a7ec581404bdb3562e64a4d70b03/dags/iop/parser.py#L134

Reply:
Yes, I know, but you should have in the loop to clean the variable after the iteration, don't you?

Reply:
I am cleaning the variable even before putting to the array, just to avoid the one more iteration

Reply:
It cleans, because even if the surname is not in the XML, it's assigned to None